src/pages/Categories.js

```diff
@@ -1,22 +1,43 @@
 import React, { Component } from "react";
-import ProductList from "../components/ProductList";
-import '../components/products-list.scss';
 
 class Categories extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        allProductsInCurrentCategory: [],
+        productsToDisplay: [],
+      }
+    }
+
+    componentDidMount(nextProps) {
+      console.log('Compnent did mount')
+      const { pathContext } = this.props;
+      this.setState({
+        allProductsInCurrentCategory: pathContext.products,
+        productsToDisplay: pathContext.products, 
+      });
+    }
+
+    applyFilter  = () => {
+      const { allProductsInCurrentCategory } = this.state;
+      const filteredProducts = [allProductsInCurrentCategory[0], allProductsInCurrentCategory[1], allProductsInCurrentCategory[2]];
+      this.setState({
+        productsToDisplay: filteredProducts
+      });
+    }
 
     render() {
         const { pathContext } = this.props;
         return (
-            <div className="container demo-container">
-                <div className="row">
-                    <div className="demo-product-collection">
-                        <div className="demo-product-collection-header">
-                            <p>{`${pathContext.productType} collection`}</p>
-                        </div>
-                        <ProductList products={this.props.pathContext.products} />
-                    </div>
-                </div>
-            </div>
+          <React.Fragment>
+            {
+              this.props.renderCategoriesListPage({
```

Comment:
Here we are using render prop to render actual content. Additionally, all the data needed form the page is sent as arguments to this`renderCategoriesListPage` functions. We can also pass other functions to like how I have passed `applyFilter` to be used by the theme component. So those who write theme component can make use of these data and functions and build the component.